scripts/go-build-ldflags.sh

```diff
@@ -9,10 +9,17 @@ fi
 
 export GIT_REVISION=$(git rev-parse --short HEAD)
 export GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-export GIT_VERSION=$(git describe --tags --abbrev=0 2>/dev/null || echo unknown)
 export BUILD_DATE=$(date '+%F-%T') # outputs something in this format 2017-08-21-18:58:45
 export BUILD_TS_UNIX=$(date '+%s') # second since epoch
 export BASE_PACKAGE=github.com/uber/cadence/common/metrics
+if [ -z ${SERVER_VERSION} ]; then
+  # If not set SERVER_VERSION, then use the most recent tag.
+  export GIT_VERSION=$(git describe --tags --abbrev=0 2>/dev/null || echo unknown)
+else
+  # If passing a version explicitly, then use it
+  export GIT_VERSION=${SERVER_VERSION}
+fi
+
```

Comment:
Two Qs / options:
1. Maybe include `--dirty` so you see if the git status is clean?  I've found a fair number of flawed build systems this way, since it'll tell you if you're un-knowingly modifying things in your releases.
2. If ya want to be fancy:
    ```bash
    export GIT_VERSION="${SERVER_VERSION:-$(git describe --tags --abbrev=0 2>/dev/null || echo unknown)}"
    ```
    it'll only run the `$(...)` operation if SERVER_VERSION is not set.  Arguably this could be done for all the vars, though I mostly doubt we'll ever want to customize the others.
    
    (I generally avoid fancy bash-isms, but some are simple / clear / common enough to be useful)

But this is good as-is 👍 Same as before, and unambiguous.

Reply:
adopted 1 but reserve for 2 -- probably I am not very good at bash and feel like less fancyness will be more readable for people like me :P 
(really wish some we could write all python for all script).

Reply:
yep, more-explicit bash is rarely the wrong choice I think :)  easier to read and modify later.